kratos/processes/variational_distance_calculation_process.h

```diff
@@ -451,81 +424,61 @@ class VariationalDistanceCalculationProcess
         mp_distance_model_part->SetProperties(base_model_part.pProperties());
         mp_distance_model_part->Tables() = base_model_part.Tables();
 
-        //assigning the nodes to the new model part
+        // Assigning the nodes to the new model part
         mp_distance_model_part->Nodes() = base_model_part.Nodes();
 
-        //ensure that the nodes have distance as a DOF
-        for (ModelPart::NodesContainerType::iterator iii = base_model_part.NodesBegin(); iii != base_model_part.NodesEnd(); iii++)
-        {
-            iii->AddDof(DISTANCE);
-        }
+        // Ensure that the nodes have distance as a DOF
+        VariableUtils().AddDof<Variable<double> >(DISTANCE, base_model_part);
 
-        //generating the elements
+        // Generating the elements
         mp_distance_model_part->Elements().reserve(base_model_part.Elements().size());
-        for (ModelPart::ElementsContainerType::iterator iii = base_model_part.ElementsBegin(); iii != base_model_part.ElementsEnd(); iii++)
-        {
-            Properties::Pointer properties = iii->pGetProperties();
-            Element::Pointer p_element = Element::Pointer(new DistanceCalculationElementSimplex<TDim>(
-                                             iii->Id(),
-                                             iii->pGetGeometry(),
-                                             iii->pGetProperties() ) );
-
-            //assign EXACTLY THE SAME GEOMETRY, so that memory is saved!!
-            p_element->pGetGeometry() = iii->pGetGeometry();
+        for (auto it_elem = base_model_part.ElementsBegin(); it_elem != base_model_part.ElementsEnd(); ++it_elem){
+            Properties::Pointer properties = it_elem->pGetProperties();
+            Element::Pointer p_element = Kratos::make_shared<DistanceCalculationElementSimplex<TDim> >(
+                it_elem->Id(),
+                it_elem->pGetGeometry(),
+                it_elem->pGetProperties());
+
+            // Assign EXACTLY THE SAME GEOMETRY, so that memory is saved!!
+            p_element->pGetGeometry() = it_elem->pGetGeometry();
             
             mp_distance_model_part->Elements().push_back(p_element);
```

Comment:
Also not done by you, but I thought we should not bypass the addition of elements to a MP?
@RiccardoRossi